helloworld/client.go

```diff
@@ -17,54 +17,51 @@ package helloworld
 
 import (
 	"context"
+	"fmt"
 
 	trillian "github.com/google/trillian"
 	"github.com/google/trillian/merkle/logverifier"
 	"github.com/google/trillian/merkle/rfc6962/hasher"
 
 	"github.com/google/trillian-examples/formats/log"
+	"golang.org/x/mod/sumdb/note"
 )
 
 // Personality is the front-end for the Trillian log.
 type Personality interface {
 	// Append stores an entry in the log and, once it is there, returns a
 	// new checkpoint that commits to the new entry (in addition to all
 	// previous ones).
-	Append(context.Context, []byte) (*log.Checkpoint, error)
+	Append(context.Context, []byte) ([]byte, error)
```

Comment:
I would like to see an alias for `[]byte` here to some new type like `RawCheckpoint` (it's not obvious at a first glance at this new sig that the input bytes are leaf preimage, and the output bytes are a raw checkpoint). We could later do the same with `RawProof`. I think it makes it easier to read and avoids type errors. A side bonus is potential for custom methods on there, but even if yagni, the other reasons motivate me.

Reply:
Sure, can give it a go - but do you actually mean a type alias, or did you mean a named type definition of a `[]byte`?

In practice I'm not sure it'll actually help all that much with type errors (e.g. `note.Open` still takes a `[]byte`, and on-disk and over the wire representations are `[]byte`, etc.), and personally I still find redefinitions of primitive types to be a bit misleading ("yay, I'll just grab the sigs from this struct"... "huh?" ... "oh, it's a `[]byte` :(") but maybe I'm a dinosaur 🦖  and I guess if nothing else it's a bit more documentation.

I've gone with type definition for now, PTAL and see what you think.